src/openrct2/world/Footpath.cpp

```diff
@@ -514,65 +515,65 @@ static TileElement* footpath_connect_corners_get_neighbour(int32_t x, int32_t y,
+                            currentPos };
+        if (tileElements[2].first == nullptr)
             continue;
 
-        direction = (direction + 1) & 3;
-        x += CoordsDirectionDelta[direction].x;
-        y += CoordsDirectionDelta[direction].y;
+        direction = direction_next(direction);
+        currentPos += CoordsDirectionDelta[direction];
         // First check link to previous tile
-        tileElement[3] = footpath_connect_corners_get_neighbour(x, y, z, (1 << direction_reverse(direction)));
-        if (tileElement[3] == nullptr)
+        tileElements[3] = { footpath_connect_corners_get_neighbour({ currentPos, z }, (1 << direction_reverse(direction))),
+                            currentPos };
+        if (tileElements[3].first == nullptr)
             continue;
         // Second check link to initial tile
-        tileElement[3] = footpath_connect_corners_get_neighbour(x, y, z, (1 << ((direction + 1) & 3)));
-        if (tileElement[3] == nullptr)
+        tileElements[3] = { footpath_connect_corners_get_neighbour({ currentPos, z }, (1 << ((direction + 1) & 3))),
+                            currentPos };
+        if (tileElements[3].first == nullptr)
             continue;
 
-        direction = (direction + 1) & 3;
-        tileElement[3]->AsPath()->SetCorners(tileElement[3]->AsPath()->GetCorners() | (1 << (direction)));
-        map_invalidate_element({ x, y }, tileElement[3]);
+        direction = direction_next(direction);
+        tileElements[3].first->SetCorners(tileElements[3].first->GetCorners() | (1 << (direction)));
+        map_invalidate_element(tileElements[3].second, reinterpret_cast<TileElement*>(tileElements[3].first));
 
-        direction = (direction - 1) & 3;
-        tileElement[2]->AsPath()->SetCorners(tileElement[2]->AsPath()->GetCorners() | (1 << (direction)));
+        direction = direction_prev(direction);
+        tileElements[2].first->SetCorners(tileElements[2].first->GetCorners() | (1 << (direction)));
 
-        map_invalidate_element({ x, y }, tileElement[2]);
+        map_invalidate_element(tileElements[2].second, reinterpret_cast<TileElement*>(tileElements[2].first));
 
-        direction = (direction - 1) & 3;
-        tileElement[1]->AsPath()->SetCorners(tileElement[1]->AsPath()->GetCorners() | (1 << (direction)));
+        direction = direction_prev(direction);
+        tileElements[1].first->SetCorners(tileElements[1].first->GetCorners() | (1 << (direction)));
 
-        map_invalidate_element({ x, y }, tileElement[1]);
+        map_invalidate_element(tileElements[1].second, reinterpret_cast<TileElement*>(tileElements[1].first));
 
-        direction = initialDirection;
-        tileElement[0]->AsPath()->SetCorners(tileElement[0]->AsPath()->GetCorners() | (1 << (direction)));
-        map_invalidate_element({ x, y }, tileElement[0]);
+        direction = direction_prev(direction);
```

Comment:
Its the same value but can you make this the same as the original i.e. set it to initialDirection.